packages/block-editor/src/hooks/anchor.js

```diff
@@ -91,7 +86,7 @@ function BlockEditAnchorControlPure( { anchor, setAttributes } ) {
 				onChange={ ( nextValue ) => {
 					nextValue = nextValue.replace( ANCHOR_REGEX, '-' );
 					setAttributes( {
-						anchor: nextValue,
+						anchor: nextValue !== '' ? nextValue : undefined,
```

Comment:
This is necessary to avoid storing an empty string in comment delimiters. Also, without this change, some unit tests would fail.